src/components/Card.js

```diff
@@ -1,53 +1,77 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
-    this._name = data.name;
-    this._link = data.link;
+  constructor(
+    cardData,
+    cardSelector,
+    handlePreviewPicture,
+    handleDeleteClick,
+    handleLikeClick
+  ) {
     this._cardSelector = cardSelector;
-    this._handleImageClick = handleImageClick;
-  }
-
-  _getTemplate() {
-    const cardElement = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
-    return cardElement;
+    this._name = cardData.name;
+    this._link = cardData.link;
+    this._handlePreviewPicture = handlePreviewPicture;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
+    this._cardId = cardData._id;
+    this._isLiked = cardData.isLiked;
   }
 
   _setEventListeners() {
-    this._likeButton.addEventListener("click", () => {
-      this._handleLikeIcon();
+    this._cardElement
+      .querySelector(".card__like-button")
+      .addEventListener("click", () => {
+        this._handleLikeClick(this);
+      });
+    this._cardElement
+      .querySelector(".card__trash-button")
+      .addEventListener("click", () => {
+        this._handleDeleteClick(this);
+      });
+    this._cardImageEl.addEventListener("click", () => {
+      this._handlePreviewPicture({ name: this._name, link: this._link });
     });
+  }
 
-    this._trashButton.addEventListener("click", () => {
-      this._handleDeleteCard();
-    });
+  renderLikes(isLiked) {
+    this._isLiked = isLiked;
+    const likeButton = this._cardElement.querySelector(".card__like-button");
```

Comment:
const likeButton = this._cardElement.querySelector(".card__like-button");

Reply:
Hello! Thank you for the feedback. I'm not sure what this comment means. Can you clarify?

Reply:
> Hello! Thank you for the feedback. I'm not sure what this comment means. Can you clarify?

There was a glitch. I wanted to paste this advice

**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```